### PR TITLE
Replace jQuery with vanilla JavaScript

### LINKED ISSUE #159: switchers.js not compatible with Sphinx 6 / requires jQuery
1. Open https://docs.python.org/3.12/ and check the language and version switchers are shown in the header:

image

2. Check the footer: "Created using Sphinx 4.5.0."
3. Open https://docs.python.org/3.13/ and see the switchers are not shown: 

image

4. Check the footer: Created using Sphinx 6.2.0."
5. Check the browser console:
```
switchers.js:168 Uncaught ReferenceError: $ is not defined
    at switchers.js:168:3
    at switchers.js:182:3
```

This is because Sphinx 6 removed jQuery:

> [#7405](https://github.com/sphinx-doc/sphinx/issues/7405): Removed the jQuery and underscore.js JavaScript frameworks.
> 
> These frameworks are no longer be automatically injected into themes from Sphinx 6.0. If you develop a theme or extension that uses the `jQuery`, `$`, or `$u` global objects, you need to update your JavaScript to modern standards, or use the mitigation below.

https://www.sphinx-doc.org/en/master/changes.html#release-6-0-0-released-dec-29-2022 has a couple of options on how to re-add jQuery, but it would be better rewrite https://github.com/python/docsbuild-scripts/blob/main/templates/switchers.js to use vanilla JavaScript.

### PR DESCRIPTION
**Pull request by hugovk**
Possibly fixes #159.

Is there a good way to test this?

### CODE REVIEW

File: templates/switchers.js [hunk 1]
```diff
@@ -18,8 +18,8 @@
     '(?:dev)',
     '(?:release/\\d.\\d[\\x\\d\\.]*)'];
 
-  var all_versions = $VERSIONS;
-  var all_languages = $LANGUAGES;
+  var all_versions = {"3.13": "dev (3.13)", "3.12": "pre (3.12)", "3.11": "3.11", "3.10": "3.10", "3.9": "3.9", "3.8": "3.8", "3.7": "3.7", "3.6": "3.6", "3.5": "3.5", "2.7": "2.7"};
```
Comment (JulienPalard) [inline templates/switchers.js:21]: Have you hardcoded those for test purposes and forgot to not commit it?

Comment (hugovk) [reply templates/switchers.js:21]: Oops sorry, that's right, let me revert that...

**Comment by JulienPalard** [comment]
> Is there a good way to test this?

Yes, run the command in the README like:

    python3 ./build_docs.py --quick --build-root ./build_root --www-root ./www --log-directory ./logs --group $(id -g) --skip-cache-invalidation


you can limit it to one language and one version for a faster build by adding `--language en --branch main`.

**Comment by egeakman** [comment]
Ah, I almost opened a pull request for this, you were quicker :D

**Comment by hugovk** [comment]
I ran:

```
python3 ./build_docs.py --quick --build-root ./build_root --www-root ./www --log-directory ./logs --group $(id -g) --skip-cache-invalidation --language en --branch origin/main
```

I didn't get any console errors, but also didn't see the switchers:

image

I then did two languages and two versions:

```
python3 ./build_docs.py --quick --build-root ./build_root --www-root ./www --log-directory ./logs --group $(id -g) --skip-cache-invalidation --language en fr --branch 3.13 --branch 3.12
```

And served the pages via `python -m http.server -d www`, but again saw no switchers for any of the four.

**Comment by egeakman** [comment]
I can't see the switchers either in my version until I resize the page enough to see the mobile view.

image

**Comment by hugovk** [comment]
Ah right, thanks, and I had old results in the `www` directory so this time I deleted it first.


```sh
rm -rf www
python3 ./build_docs.py --quick --build-root ./build_root --www-root ./www --log-directory ./logs --group $(id -g) --skip-cache-invalidation --language en fr --branch 3.13
python3 -m http.server -d www
```


Here's `main` on the left, with console errors and no switchers; and this PR on the right with no errors and switchers:

image

**Comment by egeakman** [comment]
<select> items are missing in the switcher div

image

How it should be:

image

**Comment by hugovk** [comment]
Okay, looks like it's working now at the top:

image

And in the mobile sidebar:

image

There was a problem that `$('.version_switcher_placeholder')` returned all placeholders, but `document.querySelector('.version_switcher_placeholder');` returned only the first. So we need to use `querySelectorAll` and loop.

There was also a possible with adding too many quotes near the `quote_attr` calls.

**Review by JulienPalard — APPROVED** [review]

**Comment by JulienPalard** [comment]
I'm very happy to see this land, thanks for the work!

**Pull request merged**